web/js/waste.js

```diff
@@ -75,7 +77,7 @@ $(function() {
           $('#new_bin_count').text(0);
       }
       $('#apply_discount_wrapper').hide();
-      if ($('[name=apply_discount]')[0] && $('[name=apply_discount]')[0].checked) {
+      if (discount_selected) {
```

Comment:
Presumably the `$('#modify [name=apply_discount]')` line below is no longer needed if the tick box is no longer present.